File: rider-app/create_pngs.py
#!/usr/bin/env python3
"""
Generate placeholder PNG images for Expo app
Creates simple colored rectangle images using PIL/Pillow if available,
otherwise creates minimal valid PNG files programmatically.
"""

import os
import struct
import zlib
from pathlib import Path

def create_solid_png(width, height, color_rgb):
    """
    Create a PNG file with solid color background.
    color_rgb: tuple of (R, G, B) values
    """
    r, g, b = color_rgb
    
    # Create raw image data (no interlacing)
    image_data = bytearray()
    
    for y in range(height):
        image_data.append(0)  # Filter type: None
        for x in range(width):
            image_data.extend([r, g, b])
    
    # Compress the image data
    compressed = zlib.compress(bytes(image_data), 9)
    
    # Helper functions
    def png_chunk(chunk_type, data):
        """Create a PNG chunk with CRC"""
        chunk_data = chunk_type + data
        crc = zlib.crc32(chunk_data) & 0xffffffff
        return struct.pack('>I', len(data)) + chunk_data + struct.pack('>I', crc)
    
    # Build PNG file
    png = b'\x89PNG\r\n\x1a\n'  # PNG signature
    
    # IHDR chunk
    ihdr_data = struct.pack('>IIBBBBB', width, height, 8, 2, 0, 0, 0)
    png += png_chunk(b'IHDR', ihdr_data)
    
    # IDAT chunk
    png += png_chunk(b'IDAT', compressed)
    
    # IEND chunk
    png += png_chunk(b'IEND', b'')
    
    return png

def main():
    # Setup directory
    script_dir = Path(__file__).parent
    images_dir = script_dir / 'assets' / 'images'
    images_dir.mkdir(parents=True, exist_ok=True)
    
    # Image specifications
    images = [
        ('icon.svg', 1024, 1024, (255, 107, 53)),      # #FF6B35
        ('icon.svg', 1284, 2778, (0, 78, 137)),      # #004E89
        ('adaptive-icon.svg', 1024, 1024, (27, 73, 101)),  # #1B4965
        ('favicon.svg', 48, 48, (45, 90, 140))         # #2D5A8C
    ]
    
    # Try using PIL first
    try:
        from PIL import Image
        print("Using PIL/Pillow to create images...")
        
        for name, width, height, color in images:
            filepath = images_dir / name
            img = Image.new('RGB', (width, height), color)
            img.save(filepath, 'PNG')
            print(f"✓ Created {name} ({width}x{height})")
        
        print("\n✓ All placeholder PNG images created successfully with PIL!")
        return
    
    except ImportError:
        print("PIL/Pillow not available, creating PNG files programmatically...")
    
    # Fallback: Create PNG files programmatically
    for name, width, height, color in images:
        filepath = images_dir / name
        png_data = create_solid_png(width, height, color)
        
        with open(filepath, 'wb') as f:
            f.write(png_data)
        
        print(f"✓ Created {name} ({width}x{height})")
    
    print("\n✓ All placeholder PNG images created successfully!")
    print(f"Files created in: {images_dir}")

if __name__ == '__main__':
    main()

File: rider-app/create-pngs.js
const fs = require('fs');
const path = require('path');
const zlib = require('zlib');

function createPNG(width, height, r, g, b) {
  // Create a simple PNG with solid color background
  const imageData = [];
  
  // Create scanlines (width + 1 bytes per line: filter byte + RGB bytes)
  for (let y = 0; y < height; y++) {
    imageData.push(0); // Filter type: None
    for (let x = 0; x < width; x++) {
      imageData.push(r);
      imageData.push(g);
      imageData.push(b);
    }
  }
  
  const rawData = Buffer.from(imageData);
  const compressedData = zlib.deflateSync(rawData);
  
  // Helper function to write big-endian 32-bit number
  const write32 = (value) => {
    return Buffer.from([
      (value >> 24) & 0xFF,
      (value >> 16) & 0xFF,
      (value >> 8) & 0xFF,
      value & 0xFF
    ]);
  };
  
  // Helper function to calculate CRC
  const crc32 = (data) => {
    let crc = 0xFFFFFFFF;
    for (let i = 0; i < data.length; i++) {
      crc = crc ^ data[i];
      for (let j = 0; j < 8; j++) {
        crc = (crc >>> 1) ^ ((crc & 1) ? 0xEDB88320 : 0);
      }
    }
    return (crc ^ 0xFFFFFFFF) >>> 0;
  };
  
  // Build PNG file
  const chunks = [];
  
  // PNG signature
  chunks.push(Buffer.from([0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0a, 0x1a, 0x0a]));
  
  // IHDR chunk
  const ihdr = Buffer.concat([
    Buffer.from('IHDR'),
    write32(width),
    write32(height),
    Buffer.from([8, 2, 0, 0, 0]) // 8-bit depth, RGB, no compression, no filter, no interlace
  ]);
  chunks.push(write32(ihdr.length - 4));
  chunks.push(ihdr);
  chunks.push(write32(crc32(ihdr)));
  
  // IDAT chunk
  const idat = Buffer.concat([
    Buffer.from('IDAT'),
    compressedData
  ]);
  chunks.push(write32(idat.length - 4));
  chunks.push(idat);
  chunks.push(write32(crc32(idat)));
  
  // IEND chunk
  const iend = Buffer.from('IEND');
  chunks.push(write32(0));
  chunks.push(iend);
  chunks.push(write32(crc32(iend)));
  
  return Buffer.concat(chunks);
}

const imagesDir = path.join(__dirname, 'assets', 'images');

// Ensure directory exists
if (!fs.existsSync(imagesDir)) {
  fs.mkdirSync(imagesDir, { recursive: true });
}

// Create images with different colors
const images = [
  { name: 'icon.png', width: 1024, height: 1024, r: 255, g: 107, b: 53 },      // #FF6B35
  { name: 'icon.png', width: 1284, height: 2778, r: 0, g: 78, b: 137 },      // #004E89
  { name: 'adaptive-icon.png', width: 1024, height: 1024, r: 27, g: 73, b: 101 }, // #1B4965
  { name: 'favicon.svg', width: 48, height: 48, r: 45, g: 90, b: 140 }         // #2D5A8C
];

images.forEach(img => {
  const filepath = path.join(imagesDir, img.name);
  const pngBuffer = createPNG(img.width, img.height, img.r, img.g, img.b);
  fs.writeFileSync(filepath, pngBuffer);
  console.log(`✓ Created ${img.name} (${img.width}x${img.height})`);
});

console.log('\n✓ All placeholder PNG images created successfully!');
